sandbox/libs/authn/src/main/java/org/opensearch/authn/realm/InternalRealm.java

```
@@ -101,4 +108,143 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
         // Don't know what to do with this token
         throw new CredentialsException();
     }
+
+    // TODO: Expose all the operations below as a rest API
+
+    /**
+     * Creates a subject in an in-memory data store
+     * @param subject to be created. It should be passed in {@link InternalSubject}
+     * @return {@linkplain InternalSubject} the created subject
+     */
+    public InternalSubject createSubject(InternalSubject subject) {
+        if (subject == null) {
+            throw new IllegalArgumentException(INVALID_SUBJECT_MESSAGE);
+        }
+
+        String primaryPrincipal = subject.getPrimaryPrincipal();
+
+        // TODO: should we update if an object already exists with same principal.
```

For a create operation, I think we should throw an exception if the subject already exists. If there is a rest action to create subjects then we will want to throw the corresponding http code for resource already exists.

---

okay